wire-swift-generator/src/main/java/com/squareup/wire/swift/SwiftGenerator.kt

```diff
@@ -498,6 +532,12 @@ class SwiftGenerator private constructor(
               }
             }
             decoder.add(")")
+
+            // local proto3 enum fields are initialized to a default value, so ensure we keep that if the decode returns nil
+            if (type.syntax == PROTO_3 && field.isEnum && !field.isRepeated) {
+              decoder.add(" ?? %N", field.name)
```

Comment:
This feels dangerous. IIRC, this is for the case where we return `nil` for an unknown enum case, but in this case we're going to fill that in with the default value. If that's the case, then I think this is still an error (at least IMO it is). If you want the "return nil" behavior, then you can/should mark the field as optional (which still works in proto3).

Reply:
The only alternative is to force unwrap the `decode` and crash if the enum decoding strategy is set to `.returnNil`.  The local `type` here isn't optional:

```diff
extension Person3.PhoneNumber : Proto3Codable {
    public init(from reader: ProtoReader) throws {
        var number: String = ""
-        var type: Person3.PhoneType = .MOBILE
+        var type: Person3.PhoneType = Person3.PhoneType(rawValue: 0)!

        let token = try reader.beginMessage()
        while let tag = try reader.nextTag(token: token) {
            switch tag {
            case 1: number = try reader.decode(String.self)
-            case 2: type = try reader.decode(Person3.PhoneType.self) ?? type
+            case 2: type = try reader.decode(Person3.PhoneType.self)!
            default: try reader.readUnknownField(tag: tag)
            }
        }
        self.unknownFields = try reader.endMessage(token: token)

        self.number = number
        self.type = type
    }
```

Some much force unwrapping makes me think it's the wrong approach.  If folks are worried about correctness, wouldn't they use the `throwError` strategy?

Reply:
Why not: 
```swift
var type: Person3.PhoneType?
// ...
case 2: type = try reader.decode(Person3.PhoneType.self)
//
self.type = try {
    guard let type = type ?? Person3.PhoneType(rawValue: 0) else {
        throw SomeError()
    }
    return type
}()
```

Reply:
I like that approach, I'll add an extension that can be used similarly to `checkIfMissing()`

Reply:
Can't we check this at codegen time (and refuse to generate if it's wrong)? We know at that point whether or not case 0 exists.

Reply:
The proto structure itself is checked in https://github.com/square/wire/blob/master/wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/SyntaxRules.kt#L104, but I added a runtime check that will protect against pruning the 0th value